core/src/main/java/io/undertow/client/http/HttpClientExchange.java

```diff
@@ -122,6 +122,9 @@ void setContinueResponse(ClientResponse response) {
     void setResponse(ClientResponse response) {
         this.response = response;
         if (responseCallback != null) {
+            try {
+                Thread.sleep(500);
```

Comment:
We can't add a sleep to a core class because of a test, only to the test itself.